packages/runtime/src/public/createApp.ts

```diff
@@ -0,0 +1,126 @@
+// Note: We should be careful of what we import in this module, we don't want to load the entire runtime here
+// until createApp is called, since we set some globals here, which will be read by other runtime modules
+import '@mobilestack-xyz/runtime/src/missingGlobals'
+import Config from 'react-native-config'
+import DeviceInfo from 'react-native-device-info'
+import { ToggleableOnboardingFeatures } from '../onboarding/types'
+
+// This will evolve. We should be mindful of breaking changes.
+// This structure should scale well as we add more features
+// and makes it clear what's core configuration vs optional features.
+//
+// Guidelines:
+// - We should only have a few core configuration options, and the rest should be optional features and/or with default values
+// - We should only add new configuration options that we want to support long term, and not just for a specific app
+// - Configuration options should be well documented and have clear purposes
+// - Breaking changes to configuration should be avoided when possible
+// - Configuration should be type-safe. In some cases we can consider runtime validation.
+interface PublicAppConfig {
+  registryName: string
+  displayName: string
+  deepLinkUrlScheme: string
+
+  // Platform specific configuration
+  ios?: {
+    appStoreId?: string
+  }
+
+  // Theme configuration
+  themes?: {
+    // Rough example of what we might want to support
+    default: {
+      // To adjust status bar style, keyboard appearance, etc
+      isDark?: boolean
+      colors?: {
+        // Core brand colors
+        primary?: string
+        secondary?: string
+        background?: string
+
+        // Semantic colors
+        error?: string
+        success?: string
+        warning?: string
+
+        // Text colors
+        text?: {
+          primary?: string
+          secondary?: string
+          disabled?: string
+        }
+      }
+    }
+  }
+
+  // Optional features/capabilities
+  features?: {
+    firebase?: boolean
+    sentry?: {
+      clientUrl: string
+    }
+    // TODO: what's the marketing name for this?
+    cloudBackup?: boolean
+    onboarding?: {
+      enableBiometry?: boolean
+      protectWallet?: boolean
+    }
+    walletConnect?: {
+      projectId: string
+    }
+  }
+
```

Comment:
Perhaps one other candidate for this features config would be the [quick action](https://github.com/valora-inc/wallet/blob/main/src/config.ts#L213) selection we currently have - it seems like a good fit here.

Reply:
log level might be good to add as well

Reply:
Yes indeed, I'm taking note and will propose adding as follow up PRs.